crates/solver/src/settlement_submission/submitter.rs

```diff
@@ -65,6 +65,14 @@ pub struct TransactionHandle {
     pub tx_hash: H256,
 }
 
+#[derive(Debug, Clone)]
+pub struct CancelHandle {
+    /// transaction previosly submitted using TransactionSubmitting::submit_transaction()
+    pub submitted_transaction: TransactionHandle,
+    /// empty transaction with the same nonce used for cancelling the previously submitted transaction
+    pub noop_transaction: TransactionBuilder<DynTransport>,
```

Comment:
Not a huge fan of having this "leak" to the `submitter` (to me the fact that a strategy uses a "no-op transaction" is an implementation detail of that strategy).

That being said, this would require the submitter to keep additional information (nonce, last gas price used, etc.) so I don't see a very good way around this unfortunately.

Reply:
True. But at least I don't see a third way of cancelling the tx than these two (in the future), so this structure can be treated as sealed.